src/main/java/bhs/devilbotz/RobotContainer.java

```diff
@@ -162,7 +155,9 @@ private void configureBindings() {
     SmartDashboard.putData("gripperClose", new GripperClose(gripper));
 
     SmartDashboard.putData(
-        "armScorePiece", new ArmMoveDistance(arm, -10).andThen(new GripperOpen(gripper)));
+        "armScorePiece",
+        new ArmMoveDistance(arm, ArmConstants.POSITION_SCORING_DELTA, gripper)
+            .andThen(new GripperOpen(gripper)));
```

Comment:
i don't think we want gripper to open here.  The button assigned this command does not open gripper.  The dashboard button should do the same thing.  That said, we don't really need this put to dashboard since we put arm commands onto their own shuffleboard tab in buildArmShuffleboardTab()

Reply:
> These changes are huge. And not just adding debug messages. I don't like that we are changing so much last minute. We should be changing code as little as possible now. Focus on letting drivers practice and fix things as we find during driver practice. I don't agree with merging this PR now.

Hi Maciej,

The changes do look "huge" in terms of line change count, but logically the code is much simpler.  The debug code did find a bunch of bugs and corner cases during simulation runs.  The bug fixes are currently interspersed with the debug logic, so I'll split these into two PRs so that it's cleaner.

> Test mode is not going to be useful for this. We want to be able to trace/watch command actions during matches, not just in test mode. Test mode and live window was intended to be a way for teams to debug individual components on their robot without any of their own code running. This would let you verify that wiring was good, parts were good.. but it's kind of fallen out of favor for reasons. Not all vendors don't support it, hard to keep up with new components coming out. More background here: https://www.chiefdelphi.com/t/have-you-used-the-driver-station-test-mode-successfully-this-year/428685
> 
> As an aside, there are already solutions written by FRC community we should look at, two that come to mind are Oblog and Advantage Kit/Scope:
> 
> * https://www.chiefdelphi.com/t/oblog-2023-beta-release/417429
> * https://www.chiefdelphi.com/t/introducing-logging-with-advantagekit/398612
> * https://www.chiefdelphi.com/search?q=oblog
> 
> I played with Oblog a bit last year, never tried Advantage stuff.

I did see the Advantage/Ooblog stuff and it did seem useful, but for tomorrow's testing, I needed something simple.  Nothing beats good ol' console prints for debugging realtime systems.  I'm OK keeping this debug stuff in a branch.  As I mentioned above, I'll push the bug fixes as a separate PR.

Reply:
This is what was in mainline.  I just changed it from using the "-10" to the constant and the gripper object for the "auto close" if moving down too close to the frame.  I agree we need to remove a lot of the SmartDashboard calls that have been sprinkled about the code and come up with a more consistent debug methodology.